internal/events/aggregator.go

```diff
@@ -183,6 +183,20 @@ func (ag *aggregator) getPins(ctx context.Context, filter database.Filter) ([]ff
 	return ls, err
 }
 
+func (ag *aggregator) extractBatchMessagePin(batch *fftypes.Batch, requiredIndex int64) (totalBatchPins int64, msg *fftypes.Message, msgBaseIndex int64) {
```

Comment:
The re-write of this logic, is the main thing here.

We trigger from the `pin` - which references the `batch` and the `index`, but we need two other bits of information:
- Which message within the batch lines up with this index
   - Noting that each message might have multiple topics
- What the first index is of this message in the batch
   - This is to account for the new logic from #483 that ensures all pins associated with a message get marked `dispatched=true` together